mailpoet/assets/js/src/segments/dynamic/dynamic_segments_filters/fields/days_period_field.tsx

```diff
@@ -17,12 +18,19 @@ export function DaysPeriodField({ filterIndex }: FilterProps): JSX.Element {
     (select) => select(storeName).getSegmentFilter(filterIndex),
     [filterIndex],
   );
-  const { updateSegmentFilterFromEvent } = useDispatch(storeName);
+  const { updateSegmentFilterFromEvent, updateSegmentFilter } =
+    useDispatch(storeName);
+
+  if (!['inTheLast', 'allTime'].includes(segment.timeframe)) {
```

Comment:
Maybe we could define constants for these values ('inTheLast', 'allTime') the same way as we [do for email action types](https://github.com/mailpoet/mailpoet/blob/c0df921998a91d647b17c85bcc7476dbdef38f43/mailpoet/assets/js/src/segments/dynamic/dynamic_segments_filters/email.tsx#L21-L32) so that we lower a chance of an error due a typo.